http/v1.md

```diff
@@ -0,0 +1,493 @@
+  *   makerAssetAddress [string] returns orders where the contract address for the maker asset matches the value specified
+  *   takerAssetAddress [string] returns orders where the contract address for the taker asset matches the value specified
+
+Order specific parameters:
+  *   exchangeAddress [string]: returns orders created for this exchange address
+  *   senderAddress [string]: returns orders with the specified senderAddress
+  *   makerAssetData [string]: returns orders with the specified makerAssetData
+  *   takerAssetData [string]: returns orders with the specified takerAssetData
+  *   traderAssetData [string]: returns orders where either makerAssetData or takerAssetData has the value specified
+  *   makerAddress [string]: returns orders with the specified makerAddress
+  *   takerAddress [string]: returns orders with the specified takerAddress
+  *   traderAddress [string]: returns orders where either makerAddress or takerAddress has the value specified
+  *   feeRecipientAddress [string]: returns orders where feeRecipientAddress is feeRecipient address
+
+All parameters are optional.
+
+If both makerAssetData and takerAssetData are specified, returned orders will be sorted by price determined by (takerTokenAmount/makerTokenAmount) in ascending order. By default, orders returned by this endpoint are unsorted.
+
+While there is some redundancy in supporting maker/takerAssetType, maker/takerAssetAddress, and maker/takerAssetData, they are actually all needed. For example, you cannot query "all ERC712 orders", or "all Cryptokitties orders", or "all ERC20 orders" with just maker/takerAssetData and need the other query params to do so.
+
+#### Response
+
+[See response schema](https://github.com/0xProject/0x-monorepo/blob/development/packages/json-schemas/schemas/signed_orders_schema.ts#L1)
+
+
+```
+{
+    "total": 984,
+    "page": 1,
+    "perPage": 100,
+    "records": [
+        {
+          "order": {
+              "makerAddress": "0x9e56625509c2f60af937f23b7b532600390e8c8b",
+              "takerAddress": "0xa2b31dacf30a9c50ca473337c01d8a201ae33e32",
+              "feeRecipientAddress": "0xb046140686d052fff581f63f8136cce132e857da",
+              "senderAddress": "0xa2b31dacf30a9c50ca473337c01d8a201ae33e32",
+              "makerAssetAmount": "10000000000000000",
+              "takerAssetAmount": "20000000000000000",
+              "makerFee": "100000000000000",
+              "takerFee": "200000000000000",
+              "expirationTimeSeconds": "1532560590",
+              "salt": "1532559225",
+              "makerAssetData": "0xf47261b04c32345ced77393b3530b1eed0f346429d",
+              "takerAssetData": "0x0257179264389b814a946f3e92105513705ca6b990",
+              "exchangeAddress": "0x12459c951127e0c374ff9105dda097662a027093",
+              "signature": "0x012761a3ed31b43c8780e905a260a35faefcc527be7516aa11c0256729b5b351bc33"
+            },
+            "remainingFillableTakerAmount": "5000000000000000"
```

Comment:
I'm wondering if `remainingFillableTakerAmount` should be nested within an `optional` field. I wouldn't expect all relayers to include this information since technically all validation can be done client side.

Reply:
(Echoing my thoughts from the dev meeting)
Marking this as optional definitely seems like the right direction. 

It seems a little off to me to include remainingFillableTakerAmount in the SRA because in practice it shouldn't be consumed in production because the data will be unreliable coming from a relayer. Ultimately anyone using this information will have to build a way to get the correct data directly from the blockchain to launch to production. 

I definitely agree that it's a good idea to have somewhere for relayers to include any non standard information they would like to, with this as a possible example.

Reply:
@apackin thank you for participating in the discussion and yes --  the `optionals` or `meta_data` route is the one we are looking to take at the moment.